azerothcore/azerothcore-wotlk
# fix(DB): Fixed Southsea Freebooter shoots while running away in fear and while polymorphed

Fixed some flags on the smartAI of the creature so they cant shoot while fleeing or polymorphed, LOS.. etc

Closes AzerothCore issue#7013



## Changes Proposed:
-  Changed the flag of the shoot skill from 62 to 64 so they cant shoot it while fleeing, LOS, on fear etc
-  Changed the target of the fleeing skill so they only flee, they cant shoot now while fleeing.

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/7013
- Closes https://github.com/chromiecraft/chromiecraft/issues/1227

## SOURCE:

I dont know how to record video. This is a bucaneer polymorphed for more than 10 seconds without shooting.

![image](https://user-images.githubusercontent.com/87535580/126079917-488cfe9c-49de-4597-b4e9-4825884fd9a0.png)


## Tests Performed:

- Use .go c id 7856
- Hit them to low their HP and see them fleeing. They should not shoot until they go to another friendly unit.
- MAGES --> Hit them to aggro and polymorph. They should not attack


## How to Test the Changes:


1. Use .go c id 7856
2. Hit them to low their HP and see them fleeing. They should not shoot until they go to another friendly unit.
3. MAGES --> Hit them to aggro and polymorph. They should not attack


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
